datalad/support/annexrepo.py

```diff
@@ -3327,12 +3327,61 @@ def _save_add(self, files, git=None, git_opts=None):
                     if 'error-messages' in r else None,
                     logger=lgr)
 
-    def _save_post(self, message, status, partial_commit):
+    def _save_post(self, message, status, partial_commit, amend=False):
+
+        if amend and self.is_managed_branch() and \
+                self.format_commit("%B").strip() == "git-annex adjusted branch":
```

Comment:
Hard coding this should be safe :)

Annex/AdjustedBranch.hs:

```
{- This message should never be changed. -}
adjustedBranchCommitMessage :: String
adjustedBranchCommitMessage = "git-annex adjusted branch"
```

Reply:
:)

I'm always uncomfortable with that kind of thing (and I didn't look it up), but on the other hand coming up with something to match instead that may or may not be more robust against a change in annex is arbitrary too considering a sample size of 1. And actually analyzing what kind of changes the commit makes to figure its "nature" is that of an annex commit in the branch ... well, seems more expensive with no real gain in robustness.